### PR TITLE
Refactor keyboard handling into LookupMenu

### PR DESCRIPTION
**Pull request by micgro42**
Todo:
- [x] Esc handling
- [x] typing for this.$refs.menu
- [x] marking the already selected item as selected on focus
- [x] better commit message
- [x] tests

### CODE REVIEW

File: vue-components/src/components/Lookup.vue [hunk 1]
```diff
@@ -10,20 +14,16 @@
...
-			@keyup.enter.native="onEnter"
-			@keydown.tab.native="onTab"
-			@keyup.esc.native="onEsc"
 			autocomplete="off"
 		/>
 		<LookupMenu
 			class="wikit-Lookup__menu"
 			:menu-items="menuItems"
-			v-if="showMenu"
+			v-show="showMenu"
```
Comment (bereket-WMDE) [inline vue-components/src/components/Lookup.vue:22]: nice catch :+1:

File: vue-components/src/components/Lookup.vue [hunk 2]
```diff
@@ -10,20 +14,16 @@
...
 			@select="onSelect"
 			@scroll="onScroll"
-			:selected-item-index="selectedItemIndex"
```
Comment (bereket-WMDE) [inline vue-components/src/components/Lookup.vue:26]: you removed it here but it's used on the LookupMenu.vue file on line 13.
does this still work ?

Comment (micgro42) [reply vue-components/src/components/Lookup.vue:26]: I think so, the LookupMenu has it as a local state now: 
image

Comment (bereket-WMDE) [reply vue-components/src/components/Lookup.vue:26]: ahh.. I missed that one.
thanks

File: vue-components/src/components/Lookup.vue [hunk 3]
```diff
@@ -10,20 +14,16 @@
...
+			@esc="onEsc"
```
Comment (itamargiv) [inline vue-components/src/components/Lookup.vue:25]: Great name :+1:

File: vue-components/src/components/Lookup.vue [hunk 4]
```diff
@@ -52,15 +52,14 @@ import { MenuItem } from '@/components/MenuItem';
  *
  * Uses the following components internally: Input, ValidationMessage and LookupMenu
  */
-export default Vue.extend( {
+export default ( Vue as VueConstructor<Vue & { $refs: { menu: InstanceType<typeof LookupMenu> } }> ).extend( {
```
Comment (itamargiv) [inline vue-components/src/components/Lookup.vue:55]: I think we might need to find a more readable solution to this in the near future.

**Review by itamargiv — APPROVED** [review]
Well Done! works fantastically!



Leaving open until the end of the day in case anyone else wants to have a look, I also left a non blocking comment.

File: vue-components/src/components/Lookup.vue (hunk 4 above)
Comment (micgro42) [reply vue-components/src/components/Lookup.vue:55]: I think there were ideas for that in the links I read about it:

* vuejs/vue-class-component#94
* bluecanvas/vue-typed-refs

They are also in the commit message, added here for future reference

**Comment by micgro42** [comment]
> Leaving open until the end of the day in case anyone else wants to have a look

I'm sorry for merging this, I was so much on auto-pilot that I over-read this comment :facepalm:

**Pull request merged**